File: src/main/java/com/bytebreakstudios/tictactoe/rule/CannotPlayTwiceInARowRule.java
package com.bytebreakstudios.tictactoe.rule;

import com.bytebreakstudios.tictactoe.State;
import com.bytebreakstudios.tictactoe.action.PlayAction;
import com.bytebreakstudios.board.Action;
import com.bytebreakstudios.board.Rule;

public class CannotPlayTwiceInARowRule implements Rule<State> {
    @Override
    public boolean apply(State current, State next, Action action) {
        if (action instanceof PlayAction) return current.lastPlayer != next.lastPlayer;
        return true;
    }

    @Override
    public String description() {
        return null;
    }

    @Override
    public String serialize() {
        return null;
    }

    @Override
    public Rule deserialize(String s) {
        return null;
    }
}

File: src/main/scala/com/bytebreakstudios/tictactoe/scala/rule/CannotPlayTwiceInARowRule.scala
package com.bytebreakstudios.tictactoe.scala.rule

import com.bytebreakstudios.board._
import com.bytebreakstudios.tictactoe.scala.State
import com.bytebreakstudios.tictactoe.scala.action.PlayAction

object CannotPlayTwiceInARowRule extends Rule[State] {
  override def apply(cur: State, next: State, action: Action[_ <: GameBoardState]): Boolean = action match {
    case _:PlayAction => cur.lastPlayer != next.lastPlayer
    case _ => true
  }

  override def description(): String = ???

  override def serialize(): String = ???

  override def deserialize(s: String): Rule[_ <: GameBoardState] = ???
}
